src/main/java/com/conveyal/r5/analyst/cluster/AccessGridWriter.java

```diff
@@ -0,0 +1,154 @@
+package com.conveyal.r5.analyst.cluster;
+
+import com.conveyal.r5.analyst.LittleEndianIntOutputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+
+/**
+ * Random-access access grid writer - write arrays into offsets in an access grid format file (3-d array)
+ *  * Access grids look like this:
+ * Header (ASCII text "ACCESSGR") (note that this header is eight bytes, so the full grid can be mapped into a
+ *   Javascript typed array if desired)
+ * Version, 4-byte integer
+ * (4 byte int) Web mercator zoom level
+ * (4 byte int) west (x) edge of the grid, i.e. how many pixels this grid is east of the left edge of the world
+ * (4 byte int) north (y) edge of the grid, i.e. how many pixels this grid is south of the top edge of the world
+ * (4 byte int) width of the grid in pixels
+ * (4 byte int) height of the grid in pixels
+ * (4 byte int) number of values per pixel
+ * (repeated 4-byte int) values of each pixel in row major order. Values within a given pixel are delta coded.
+ */
+public class AccessGridWriter {
+    public static final Logger LOG = LoggerFactory.getLogger(GridResultAssembler.class);
+
+    /** The version of the access grids we produce */
+    public static final int VERSION = 0;
+
+    /** The version of the origins we consume */
+    public static final int ORIGIN_VERSION = 0;
+
+    /** The offset to get to the data section of the access grid file */
+    public static final long HEADER_SIZE = 9 * 4;
+
+    private BufferAbstraction buffer;
+
+    // store as longs so we can use with impunity without fear of overflow
+    private final long zoom, west, north, width, height, nSamples;
```

Comment:
It seems like we should be calling this "depth" especially since now we're using for things other than samples of travel time.